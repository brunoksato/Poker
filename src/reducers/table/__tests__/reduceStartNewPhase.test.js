const { startNewPhase } = require('../../../actions');
const { getTableById } = require('../../../selectors/tableSelector');
const reducer = require('../tableReducer');

describe('startNewPhase', () => {
  test('to match snapshot', () => {
    const state = {
      byId: {
        0: {
          id: 0,
          gameOn: true,
          seatCount: 5,
          activeSeatsCount: 5,
          button: 1,
          phase: 'flop',
          seats: [
            {
              position: 0,
              playerId: '0',
              chipsInPlay: 480,
              inHand: true,
              isAllIn: false,
              cards: ['Ac', '2h'],
              bet: 120,
            },
            {
              position: 1,
              playerId: '1',
              chipsInPlay: 0,
              inHand: true,
              isAllIn: true,
              cards: ['5c', '9h'],
              bet: 80,
            },
            {
              position: 2,
              playerId: '2',
              chipsInPlay: 200,
              inHand: false,
              isAllIn: false,
              cards: ['Ks', '2c'],
              bet: 70,
            },
            {
              position: 3,
              playerId: '3',
              chipsInPlay: 0,
              inHand: true,
              isAllIn: true,
              cards: ['Qc', '10h'],
              bet: 120,
            },
            {
              position: 4,
              playerId: '4',
              chipsInPlay: 210,
              inHand: true,
              isAllIn: false,
              cards: ['Jc', 'Jh'],
              bet: 120,
            },
          ],
          board: ['4c', '3d', 'Kd'],
        },
      },
    };
    const newState = reducer(state, startNewPhase({ tableId: 0 }));
    const table = getTableById(newState)(0);
    expect(table).toMatchSnapshot();
  });
});
