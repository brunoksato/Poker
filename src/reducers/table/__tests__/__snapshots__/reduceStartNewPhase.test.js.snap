// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`startNewPhase to match snapshot 1`] = `
Object {
  "activeSeatsCount": 5,
  "biggestBet": 0,
  "board": Array [
    "4c",
    "3d",
    "Kd",
  ],
  "button": 1,
  "gameOn": true,
  "id": 0,
  "lastPlayerToAct": null,
  "phase": "turn",
  "pot": Object {
    "1": 630,
    "3": 120,
    "current": 0,
  },
  "seatCount": 5,
  "seats": Array [
    Object {
      "bet": 0,
      "cards": Array [
        "Ac",
        "2h",
      ],
      "chipsInPlay": 480,
      "inHand": true,
      "isAllIn": false,
      "playerId": "0",
      "position": 0,
    },
    Object {
      "bet": 0,
      "cards": Array [
        "5c",
        "9h",
      ],
      "chipsInPlay": 0,
      "inHand": true,
      "isAllIn": true,
      "playerId": "1",
      "position": 1,
    },
    Object {
      "bet": 0,
      "cards": Array [
        "Ks",
        "2c",
      ],
      "chipsInPlay": 200,
      "inHand": false,
      "isAllIn": false,
      "playerId": "2",
      "position": 2,
    },
    Object {
      "bet": 0,
      "cards": Array [
        "Qc",
        "10h",
      ],
      "chipsInPlay": 0,
      "inHand": true,
      "isAllIn": true,
      "playerId": "3",
      "position": 3,
    },
    Object {
      "bet": 0,
      "cards": Array [
        "Jc",
        "Jh",
      ],
      "chipsInPlay": 210,
      "inHand": true,
      "isAllIn": false,
      "playerId": "4",
      "position": 4,
    },
  ],
  "toAct": 4,
}
`;
